dashbuilder/dashbuilder-shared/dashbuilder-js/packages/component-api/src/controller/DashbuilderComponentDispatcher.ts

```diff
@@ -56,8 +54,8 @@ export class DashbuilderComponentDispatcher implements InternalComponentDispatch
   }
 
   public init(): void {
-    Bus.INSTANCE.setListener(this.messageDispatcher);
-    Bus.INSTANCE.start();
+    this.bus.setListener(this.messageDispatcher);
+    this.bus.start();
```

Comment:
I can't use in the constructor because the componentId is given by DB when the component is initialized. What I did is move the component ID to the controller, who sends message, and it will acquire the componentId after the init and send all messages with it.

If the component id is undefined then something is really wrong with DB - and all the messages will be sent to oblivion and ignored.